edb/api/types.txt

```diff
@@ -47,4 +47,13 @@
 00000000-0000-0000-0000-000000000111 cal::relative_duration
 00000000-0000-0000-0000-000000000112 cal::date_duration
 
+00000000-0000-0000-0000-000000000120 range<std|int32>
```

Comment:
Hmm, what's the reason for static allocation for ranges?  We don't do this for arrays or tuples.

Reply:
Yeah, but in terms of how they map onto postgres, ranges are more like scalar types, and so they share scalar type's mechanism for mapping uuids to postgres type names

Reply:
This smells a bit. Can you have ranges of extension types, like PostGIS scalars? Also why does this line has this weird `std|int32` syntax?

Reply:
I agree it smells a bit, but I think it's largely making explicit an existing implicit smell?

I think if you have ranges of extension types, it would need to have its own postgres type name?

Because those are the internal type names for range types, since we escape `::` into `|` inside the type argument part of names.

Reply:
I still think there's probably no real need for static allocation here and we are simply missing a bit of indirection somewhere.  Also, IIRC static id allocation was introduced as an optimization so that clients can have pre-built codec tables, i.e. technically stable type ids are not a requirement even for built-in types.

Reply:
Pushed https://github.com/edgedb/edgedb/pull/4463/commits/2012b8f0e6bfc09160c9f7e70cb2f24e588a6d69 here removing static allocation from ranges and treating them like arrays instead.